jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildContext.java

```diff
@@ -500,7 +521,7 @@ public boolean getAlwaysCacheBaseImage() {
               targetImageConfiguration.getImageRepository(),
               baseImageConfiguration.getImageRepository(),
               httpClient)
-          .setUserAgentSuffix(getToolName());
+          .setUserAgentSuffix(getToolVersion() + " " + getToolName());
```

Comment:
Yeah this is to preserve behavior. I don't really know what to do about this, but the point of this PR was to change the injection point without changing any behavior.

jib is injected at a later point.

Reply:
But you already changed the code so that we inject `getToolVersion()` here as opposed to at a later point. My point is, can't we do the same for the `jib` string?

Reply:
Oh I see, I misread this, lemme go take a look

Reply:
So this one is a little weirder, should reigstry client never build the user agent? Perhaps it should delegate it all to an input?
https://github.com/GoogleContainerTools/jib/blob/a6aa257ddde850f66b45358b17b4d1ca4346b8b4/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java#L125

Reply:
I haven't looked into this carefully, but I agree. I do remember I thought it was wrong for `RegistryClient` to build any part of the user agent. `RegistryClient` should really be like an independent library where the libraries user should have full control over the user agent. I remember I tried to refactor this but quit shortly after it got into a snowball.